src/codeu/chat/ServerMain.java

```diff
@@ -33,7 +43,12 @@
 
 final class ServerMain {
 
+	
   private static final Logger.Log LOG = Logger.newLog(ServerMain.class);
+  
+//This is the directory where it is safe to store data across runs
+  // of the server.
+  private static File persistentPath = null;
```

Comment:
The problem I have with this has to do with scale: Let's pretend (and I tried this...) that the same computer would run multiple servers, that would be used by different people. They would obviously be using different ports, but the directory wouldn't be different. 
In the current solution, all servers share the same persistence information..